### PR TITLE
En-11896 move constants to config

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- `MaxRoundsToKeepUnprocessedMiniBlocks` and `MaxRoundsToKeepUnprocessedTransactions` constants needed to be moved to config as per [EN-11896](https://elrondnetwork.atlassian.net/browse/EN-11896)
  
## Proposed Changes
- move constants to configs
- small refactor, moved common code to `basePoolsCleaner`

## Testing procedure
- system test

### CODE REVIEW

File: process/block/poolsCleaner/miniBlocksPoolsCleaner.go [hunk 1]
```diff
@@ -29,44 +34,42 @@ type mbInfo struct {
...
+	if check.IfNil(args.MiniblocksPool) {
+		return process.ErrNilMiniBlockPool
+	}
+
+	return err
+}
+
 // StartCleaning actually starts the pools cleaning mechanism
 func (mbpc *miniBlocksPoolsCleaner) StartCleaning() {
 	var ctx context.Context
```
Comment (iulianpascalau) [inline process/block/poolsCleaner/miniBlocksPoolsCleaner.go:75]: as discussed, we might add a guard check here in order to not be able to call this method twice. The guard can be made up with a bool & mutex

Comment (sstanculeanu) [reply process/block/poolsCleaner/miniBlocksPoolsCleaner.go:84]: added

File: process/block/poolsCleaner/basePoolsCleaner.go [hunk 2]
```diff
@@ -22,13 +24,16 @@ type basePoolsCleaner struct {
 	shardCoordinator               sharding.Coordinator
 	maxRoundsToKeepUnprocessedData int64
 	cancelFunc                     func()
+	isCleaningRoutineRunning       atomic.Flag
```
Comment (iulianpascalau) [inline process/block/poolsCleaner/basePoolsCleaner.go:27]: since you have a mutex protection, could have been a plain old bool var. Can be left as it is

**Review by iulianpascalau — DISMISSED** [review]

File: process/block/poolsCleaner/basePoolsCleaner.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+// Close will close the endless running go routine
+func (base *basePoolsCleaner) Close() error {
+	base.mut.Lock()
+	defer base.mut.Unlock()
+
+	if base.cancelFunc != nil {
+		base.cancelFunc()
+		base.isCleaningRoutineRunning.Reset()
```
Comment (ssd04) [inline process/block/poolsCleaner/basePoolsCleaner.go:62]: can't we remove the mutex if it's for an atomic flag?

Comment (sstanculeanu) [reply process/block/poolsCleaner/basePoolsCleaner.go]: made the flag bool and kept mutex

File: process/block/poolsCleaner/basePoolsCleaner.go (hunk 2 above)
Comment (sstanculeanu) [reply process/block/poolsCleaner/basePoolsCleaner.go]: pushed

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            test_constants_to_config                     ##
================================================================================
- Known Warnings              6
- New Warnings                2
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================

**Pull request merged**